src/Cli/dotnet/Commands/Tool/Execute/ToolExecuteCommand.cs

```diff
@@ -25,10 +34,60 @@ internal class ToolExecuteCommand(ParseResult result) : CommandBase(result)
         private readonly bool _interactive = result.GetValue(ToolExecuteCommandParser.InteractiveOption);
         private readonly VerbosityOptions _verbosity = result.GetValue(ToolExecuteCommandParser.VerbosityOption);
         private readonly bool _yes = result.GetValue(ToolExecuteCommandParser.YesOption);
+
+        //  TODO: Does result.OptionValuesToBeForwarded work here?
+        private readonly IToolPackageDownloader _toolPackageDownloader = ToolPackageFactory.CreateToolPackageStoresAndDownloader(
+                additionalRestoreArguments: result.OptionValuesToBeForwarded(ToolExecuteCommandParser.GetCommand())).Item3;
+
+        //  TODO: Make sure to add these options to the command
+        private readonly RestoreActionConfig _restoreActionConfig = new RestoreActionConfig(DisableParallel: result.GetValue(ToolCommandRestorePassThroughOptions.DisableParallelOption),
+            NoCache: result.GetValue(ToolCommandRestorePassThroughOptions.NoCacheOption) || result.GetValue(ToolCommandRestorePassThroughOptions.NoHttpCacheOption),
+            IgnoreFailedSources: result.GetValue(ToolCommandRestorePassThroughOptions.IgnoreFailedSourcesOption),
+            Interactive: result.GetValue(ToolCommandRestorePassThroughOptions.InteractiveRestoreOption));
+
+        //  TODO: Use prerelease
         private readonly bool _prerelease = result.GetValue(ToolExecuteCommandParser.PrereleaseOption);
 
+        private readonly ToolManifestFinder _toolManifestFinder = toolManifestFinder ?? new ToolManifestFinder(new DirectoryPath(currentWorkingDirectory ?? Directory.GetCurrentDirectory()));
+
         public override int Execute()
         {
+            VersionRange versionRange = _parseResult.GetVersionRange();
+            PackageId packageId = new PackageId(_packageToolIdentityArgument.Id);
+
+            //  Look in local tools manifest first, but only if version is not specified
+            if (versionRange == null)
```

Comment:
versionRange wouldn't be null if they said something like *, right? Should we check the local tools manifest first for cases like that?

Reply:
* would mean latest version.  If any type of version is specified, whether a specific version, latest version, or whatever, we'll skip looking for a local tool.  If you ask for the latest version then I think it makes sense to look for the latest version from the NuGet sources and not use the local tool version.